share/smack/top.py

```diff
@@ -318,6 +320,7 @@ def default_clang_compile_command(args, lib = False):
   if args.memory_safety: cmd += ['-DMEMORY_SAFETY']
   if args.signed_integer_overflow: cmd += (['-ftrapv'] if not lib else ['-DSIGNED_INTEGER_OVERFLOW_CHECK'])
   if args.float: cmd += ['-DFLOAT_ENABLED']
+  if args.strings: cmd += ['-DSTRINGS_ENABLED']
```

Comment:
This compiler directive should not exist. String functions should be extracted to a separate file `share/smack/lib/string.c` which is conditionally included by the Smack scripts.